src/qibo/quantum_info/basis.py

```diff
@@ -45,33 +53,69 @@ def pauli_basis(
             f"vectorize must be type bool, but it is type {type(vectorize)} instead.",
         )
 
+    if not isinstance(sparse, bool):
+        raise_error(
+            TypeError,
+            f"sparse must be type bool, but it is type {type(sparse)} instead.",
+        )
+
     if vectorize and order is None:
         raise_error(ValueError, "when vectorize=True, order must be specified.")
 
-    basis = [matrices.I, matrices.X, matrices.Y, matrices.Z]
+    if sparse and not vectorize:
+        raise_error(
+            NotImplementedError,
+            "sparse representation is not implemented for unvectorized Pauli basis.",
+        )
+
+    basis_single = [matrices.I, matrices.X, matrices.Y, matrices.Z]
 
```

Comment:
@renatomello, maybe it would be better to create a `basis_full` in advance depending on the number of qubits and use it for both cases?
```python
if nqubits > 1:
    basis_full = list(product(basis_single, repeat=nqubits)) 
    basis_full = [reduce(np.kron, row) for row in basis_full]
else:
    basis_full = basis_single
    
if vectorize and sparse:
    basis, indexes = [], []
    for row in basis_full:
        row = vectorization(row, order=order)
        row_indexes = list(np.flatnonzero(row))
        indexes.append(row_indexes)
        basis.append(row[row_indexes])
        del row
elif vectorize and not sparse:
    basis = [vectorization(matrix, order=order) for matrix in basis_full]
else:
    basis = basis_full
```